### PR TITLE
Fix bug that caused namespaces to be dropped from XML attributes.

### PR DESCRIPTION
**Pull request by quickdudley**

### CODE REVIEW

**Review by benlangfeld — COMMENTED** [review]
Cool. Might you be able to add a test case for this as well as an entry in the changelog?

File: lib/blather/stream/parser.rb [hunk 1]
```diff
@@ -50,6 +46,10 @@ def start_element_namespace(elem, attrs, prefix, uri, namespaces)
       @namespaces[[prefix, uri]] ||= node.add_namespace(prefix, uri) if prefix && !ns_keys.include?(prefix)
       node.namespace = @namespaces[[prefix, uri]]
 
+      attrs.each do |attr|
+        node[[attr.prefix,attr.localname].select{|x|x}.join(":")] = attr.value
```
Comment (chewi) [inline lib/blather/stream/parser.rb]: It took me a moment to work out what this does. I suggest using `compact` instead.

```ruby
node[[attr.prefix, attr.localname].compact.join(":")] = attr.value
```

Comment (chewi) [reply lib/blather/stream/parser.rb]: In the interest of performance, I find this to be about 2.5x faster than what I put above:

```ruby
node["#{attr.prefix + ':' if attr.prefix}#{attr.localname}"] = attr.value
```

**Comment by singpolyma** [comment]
This seems good to me, the test does cover the right case.

**Comment by bklang** [comment]
Updated & fixed in #176

**Pull request closed without merging**